src/inmanta/server/services/resourceservice.py

```diff
@@ -1069,7 +1069,18 @@ async def get_resource_actions(
         action_id: Optional[uuid.UUID] = None,
         first_timestamp: Optional[datetime.datetime] = None,
         last_timestamp: Optional[datetime.datetime] = None,
+        exclude_changes: Optional[list[Change]] = None,
     ) -> ReturnValue[list[ResourceAction]]:
+        if exclude_changes is None:
+            exclude_changes = []
+        else:
+            try:
+                # Convert strings to Change enum
+                exclude_changes = [Change(change) for change in exclude_changes]
```

Comment:
Why do we need this conversion ? isn't `exclude_changes` already a `list[Change]` ?

Reply:
the typing in the arguments is wrong, currently those are strings. ([see arnauds comment](https://github.com/inmanta/inmanta-core/pull/6837#discussion_r1426564804))